doc/news/changes.h

```diff
@@ -495,6 +495,14 @@ inconvenience this causes.
 <h3>Specific improvements</h3>
 
 <ol>
+ <li> Improved: SparsityPattern::copy_from() copying from a
+ DynamicSparsityPattern argument had quadratic complexity in the number of
+ rows for sparsity patterns where most of the rows are of length zero. The bad
+ algorithm has been replaced by a linear complexity one.
+ <br>
+ (Martin Kronbichler, 2016/08/31)
```

Comment:
I know the patch is by you, but you had help finding the problem and coming up with a testcase, and I think it would be fair to acknowledge that :-)